modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala

```diff
@@ -85,10 +86,23 @@ object Enrich {
         .evalMap(chunk =>
           for {
             begin <- Clock[F].realTime(TimeUnit.MILLISECONDS)
-            result <-
-              env.semaphore.withPermit(
-                chunk.toList.map { case (orig, bytes) => enrich(bytes).map((orig, _)) }.parSequenceN(env.streamsSettings.concurrency.enrich)
-              )
+            result <- env.semaphore.withPermit(
+                        chunk.toList
+                          .map {
+                            case (orig, bytes) =>
+                              val timeout = 1.minute
+                              val badRow = genericBadRow(
```

Comment:
It seems like we are creating this bad row for every event irrespective from timeout. Might not need to change it immediately but it would be good to change it in case we want to put this feature to Enrich permanently.

Reply:
Yes good point ! We'll make it better if we decide to keep this feature (we likely will)